packages/components/src/toolbar/index.js

```diff
@@ -2,86 +2,51 @@
  * External dependencies
  */
 import classnames from 'classnames';
-import { flatMap } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+import deprecated from '@wordpress/deprecated';
 
 /**
  * Internal dependencies
  */
-import ToolbarButton from '../toolbar-button';
-import DropdownMenu from '../dropdown-menu';
+import ToolbarGroup from '../toolbar-group';
 import ToolbarContainer from './toolbar-container';
 
 /**
- * Renders a toolbar with controls.
- *
- * The `controls` prop accepts an array of sets. A set is an array of controls.
- * Controls have the following shape:
- *
- * ```
- * {
- *   icon: string,
- *   title: string,
- *   subscript: string,
- *   onClick: Function,
- *   isActive: boolean,
- *   isDisabled: boolean
- * }
- * ```
- *
- * For convenience it is also possible to pass only an array of controls. It is
- * then assumed this is the only set.
+ * Renders an accessible toolbar that follows the
+ * [WAI-ARIA Toolbar Pattern](https://www.w3.org/TR/wai-aria-practices/#toolbar).
  *
- * Either `controls` or `children` is required, otherwise this components
- * renders nothing.
+ * The `accessibilityLabel` is required. Otherwise it will produce a warning.
  *
- * @param {Object}    props
- * @param {Array}     [props.controls]  The controls to render in this toolbar.
- * @param {WPElement} [props.children]  Any other things to render inside the
- *                                      toolbar besides the controls.
- * @param {string}    [props.className] Class to set on the container div.
+ * To add controls, simply pass `ToolbarButton` components as children.
  *
- * @return {WPComponent} The rendered component.
+ * @param {Object} props
+ * @param {string} props.accessibilityLabel Required label for assistive technology users.
+ * @param {string} [props.className]
+ * @param {Object} ref
  */
-function Toolbar( { controls = [], children, className, isCollapsed, icon, label, ...otherProps } ) {
-	if (
-		( ! controls || ! controls.length ) &&
-		! children
-	) {
-		return null;
-	}
-
-	// Normalize controls to nested array of objects (sets of controls)
-	let controlSets = controls;
-	if ( ! Array.isArray( controlSets[ 0 ] ) ) {
-		controlSets = [ controlSets ];
-	}
-
-	if ( isCollapsed ) {
+const Toolbar = forwardRef( ( { className, accessibilityLabel, ...props }, ref ) => {
+	if ( accessibilityLabel ) {
 		return (
-			<DropdownMenu
-				hasArrowIndicator
-				icon={ icon }
-				label={ label }
-				controls={ controlSets }
-				className={ classnames( 'components-toolbar', className ) }
+			<ToolbarContainer
+				ref={ ref }
+				// `ToolbarGroup` already uses components-toolbar for compatibility reasons
+				className={ classnames( 'components-accessible-toolbar', className ) }
```

Comment:
I'm wondering whether we could keep the old class-name `components-toolbar` and add some modifiers to tweak the styles for the legacy use case.

Reply:
Can you explain better how it could be done?

My only concern is that the `.components-toolbar` selector is being used in other parts of Gutenberg, so I guess that other projects might also be using it as it were part of the public API.

Reply:
@jasmussen - can you help to figure out how to perform the migration for class names?

This is what happens with this PR:

1. `Toolbar` becomes `ToolbarGroup`.
2. `Toolbar` should be now the wrapper for the collection of individual toolbar buttons and/or toolbar groups.
3. @diegohaz proposed we use a `components-accessible-toolbar` class name with a very bare set of styles for the rebranded `Toolbar` component.
4. In effect, all previous occurrences fo styles need to cover both `components-toolbar` and `components-toolbar-group` class names.
5. My question is, can you think of the way, where we could repurpose the old `.components-toolbar` to use instead of `.components-accessible-toolbar`?

To better picture it, can we use a trick like:

```scss
.components-toolbar {
    // some styles for the new Toolbar
   &.legacy {
       // some old styles which get applied when using legacy version
   }
}
```

Reply:
We should avoid setting the expectation that classes are public APIs.